src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -113,9 +118,15 @@ public OAuthHandler(HttpClient backchannel)
             requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             requestMessage.Content = requestContent;
             var response = await Backchannel.SendAsync(requestMessage, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
-            return new OAuthTokenResponse(payload);
+            if (response.IsSuccessStatusCode)
+            {
+                var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+                return OAuthTokenResponse.Success(payload);
+            }
+            else
+            {
+                return OAuthTokenResponse.Failed("OAuth token endpoint failure: "+response.StatusCode);
```

Comment:
The graceful path is indeed very narrow :)  Basically we only wanted to flow targeted errors, so for this particular method, we are only trying to expose a way to do something for a bad oauth token response, nothing more.